src/main.rs

```diff
@@ -0,0 +1,124 @@
+// Require use of WinMain entrypoint so cmd doesn't flash open
+// See RFC 1665 for additional details
+#![windows_subsystem = "windows"]
+
+use winreg::RegKey;
+use std::process::{Command, exit};
+use std::io::Result;
+use std::env;
+use native_dialog::MessageDialog;
+use native_dialog::MessageType;
+use webbrowser;
+
+fn main() {
+    if let Ok(dir) = get_minecraft_installation_dir() {
+        println!("Minecraft install dir {}", dir);
+        try_minecraft_java(&dir);
+    } else {
+        println!("Could not find minecraft install dir.");
+    }
+
+    match env::var("JAVA_HOME") {
+        Ok(val) => {
+            let full_path: String = val.to_owned() + r"bin\javaw.exe";
+            launch_if_valid_java_installation(&full_path)
+        },
+        Err(_) => {},
+    }
+
+    // Getting thin on the ground, lets check the path.
+    launch_if_valid_java_installation(&"javaw".to_string());
+
+    show_error();
+}
+
+fn try_minecraft_java(dir: &String) -> bool {
+    let paths: [&str; 4] = [
+        r"runtime\jre-legacy\windows-x64\jre-legacy\bin\javaw.exe",
+        r"runtime\jre-legacy\windows-x68\jre-legacy\bin\javaw.exe",
+        r"runtime\jre-x64\bin\javaw.exe",
+        r"runtime\jre-x86\bin\javaw.exe"
+    ];
+
+    for path in &paths {
+        let full_path: String = dir.to_owned() + path;
+        launch_if_valid_java_installation(&full_path)
```

Comment:
I don't see a reason to pass the path as reference as you are going to end up dropping it after you leave the for loop.

So the parameter should take a regular `String`